pkg/iris/mapping.go

```diff
@@ -34,45 +40,92 @@ func NewMapper(
+	name       string
+	k8sVersion string
+	nodesCount int
+	osImage    string
+}
+
+// MapNodes maps a list of nodes and a given cluster name into a KubernetesObject.
+// In the process it aggregates the information from muliple nodes into one cluster object.
+func (m *mapper) GetCluster(clusterName string, kubernetesAPI *kubernetes.API) (*ClusterDTO, error) {
+
+	nodes, err := kubernetesAPI.Nodes()
 	if err != nil {
 		return nil, err
 	}
-	var allDiscoveryItems []DiscoveryItem
-	for _, namespace := range namespaces.Items {
-		// complexity is O(mxn)
-		if blacklistedNS(namespace.Name, m.BlackListedNamespaces) {
-			continue
-		}
-		log.Infof("Fetching Deployments for namespace [%s]", namespace.Name)
-		deployments, err := m.KubernetesApi.Deployments(namespace.Name)
-		if err != nil {
-			return nil, err
-		}
-		log.Infof("mapping deployments for namespace [%s]", namespace.Name)
-		mappedDeployments := m.MapDeployments(deployments)
-		allDiscoveryItems = append(allDiscoveryItems, mappedDeployments...)
+	items := nodes.Items
+	if len(items) == 0 {
+		return &ClusterDTO{
+			name: clusterName,
+		}, nil
 	}
-	log.Info("Fetching deployments for namespaces completed")
-	return allDiscoveryItems, nil
-}
+	k8sVersion := set.NewStringSet()
+	osImage := set.NewStringSet()
+
+	for _, n := range items {
+		k8sVersion.Add(n.Status.NodeInfo.KubeletVersion)
+		osImage.Add(n.Status.NodeInfo.OSImage)
 
-func (m *mapper) MapDeployments(deployments *appsv1.DeploymentList) []DiscoveryItem {
-	var deploymentDiscoveryItems []DiscoveryItem
-	for _, deployment := range deployments.Items {
-		deploymentItem := NewDeploymentEvent(*m, deployment)
-		softwareArtifactItem := NewSoftwareArtifactEvent(*m, deployment)
-		deploymentDiscoveryItems = append(deploymentDiscoveryItems, *deploymentItem, *softwareArtifactItem)
 	}
-	return deploymentDiscoveryItems
+	return &ClusterDTO{
+		name:       clusterName,
+		k8sVersion: strings.Join(k8sVersion.Items(), ", "),
+		nodesCount: len(items),
+		osImage:    strings.Join(osImage.Items(), ", "),
+	}, nil
 }
 
-func blacklistedNS(a string, list []string) bool {
-	for _, b := range list {
-		if b == a {
-			return true
+func (m *mapper) GetDeployments(namespace string, kubernetesAPI *kubernetes.API) ([]models.Deployment, error) {
+	var allDeployments []models.Deployment
+	deployments, err := kubernetesAPI.Deployments(namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	services, err := kubernetesAPI.Services(namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, deployment := range deployments.Items {
+		deploymentService := ""
+		// Check if any service has the exact same selector labels and use this as the service related to the deployment
+		for _, service := range services.Items {
+			if reflect.DeepEqual(service.Spec.Selector, deployment.Spec.Selector.MatchLabels) {
+				deploymentService = service.Name
+				break
+			}
 		}
+
+		limitCpu := deployment.Spec.Template.Spec.Containers[0].Resources.Limits[v1.ResourceCPU]
```

Comment:
Should we remove the request and limits if not specified? 
<img width="308" alt="image" src="https://user-images.githubusercontent.com/93667110/179508766-fae7435b-02c7-442d-80cf-7b0e285c2d5d.png">

Reply:
I set it to empty for now